test/topology_random_failures/cluster_events.py

```diff
@@ -0,0 +1,631 @@
+    if error_injection in (
+            "stop_after_sending_join_node_request",
+            "stop_after_bootstrapping_initial_raft_configuration",
+    ):
+        pytest.skip("See issue #18638 (rejected node is not failed to bootstrap and continue to run)")
+
+    if error_injection in (
+            "stop_after_starting_auth_service",
+            "stop_after_setting_mode_to_normal",
+            "stop_before_becoming_raft_voter",
+            "stop_after_updating_cdc_generation",
+            "stop_before_streaming",
+            "stop_after_streaming",
+    ):
+        pytest.skip("See issue #19151 (decommission process stuck while bootstrapping node is paused)")
+
+    yield
+
+    LOGGER.info("Decommission a node")
+    await manager.decommission_node(
+        server_id=(await manager.running_servers())[1].server_id,
+        timeout=TOPOLOGY_TIMEOUT,
+    )
+
+    yield
+
+
+async def remove_node(manager: ManagerClient,
+                      random_tables: RandomTables,
+                      error_injection: str) -> AsyncIterator[None]:
+    if error_injection in (
+            "stop_after_sending_join_node_request",
+            "stop_after_bootstrapping_initial_raft_configuration",
+    ):
+        pytest.skip("See issue #18638 (rejected node is not failed to bootstrap and continue to run)")
+
+    if error_injection in (
+        "stop_after_starting_auth_service",
+        "stop_after_setting_mode_to_normal",
+        "stop_before_becoming_raft_voter",
+        "stop_after_updating_cdc_generation",
+        "stop_before_streaming",
+        "stop_after_streaming",
+    ):
+        pytest.skip("Can't add a node to a cluster with a banned node")
+
+    running_servers = await manager.running_servers()
+
+    LOGGER.info("Kill a node")
+    await manager.server_stop(server_id=running_servers[1].server_id)
+    await manager.server_not_sees_other_server(
+        server_ip=running_servers[0].ip_addr,
+        other_ip=running_servers[1].ip_addr,
+    )
+
+    yield
+
+    LOGGER.info("Remove the dead node")
+    await manager.remove_node(
+        initiator_id=running_servers[0].server_id,
+        server_id=running_servers[1].server_id,
+        wait_removed_dead=False,
+        timeout=TOPOLOGY_TIMEOUT,
+    )
+
+    yield
+
+
+async def restart_non_coordinator_node(manager: ManagerClient,
+                                       random_tables: RandomTables,
+                                       error_injection: str) -> AsyncIterator[None]:
+    yield
+
+    LOGGER.info("Restart a non-coordinator node")
+    await manager.server_restart(server_id=(await manager.running_servers())[1].server_id, wait_others=3)
+
+    yield
+
+
+async def restart_coordinator_node(manager: ManagerClient,
+                                   random_tables: RandomTables,
+                                   error_injection: str) -> AsyncIterator[None]:
+    yield
+
+    LOGGER.info("Restart the coordinator node")
```

Comment:
Can we have a more reliable way to detect the raft leader here?
cc @kbr-scylla

Reply:
I think we can use this function:

https://github.com/scylladb/scylladb/blob/c11f2c9bcdffe00622a5fea0e2171d53fe361b68/test/topology/util.py#L363-L371

Reply:
Done

Reply:
@kbr-scylla ^^